File: src/main/java/nl/mpcjanssen/simpletask/sort/FileOrderComparator.java
package nl.mpcjanssen.simpletask.sort;

import com.google.common.collect.Ordering;

import org.jetbrains.annotations.NotNull;

import nl.mpcjanssen.simpletask.task.Task;

public class FileOrderComparator extends Ordering<Task> {

    @Override
    public int compare(@NotNull Task a, @NotNull Task b) {
        if (a.getId() < b.getId()) {
            return -1;
        } else {
            return 1;
        }
    }
}

File: src/main/java/nl/mpcjanssen/simpletask/sort/FileOrderComparator.kt
package nl.mpcjanssen.simpletask.sort

import nl.mpcjanssen.simpletask.dao.gen.TodoItem
import java.util.*


class FileOrderComparator  : Comparator<TodoItem> {
    override fun compare(a: TodoItem?, b: TodoItem?): Int {
        if (a === b) {
            return 0
        } else if (a == null) {
            return -1
        } else if (b == null) {
            return 1
        }
        return (a.line - b.line).toInt()
    }
}